datastore/api/snippets.py

```diff
@@ -0,0 +1,823 @@
+        'description': 'Learn Cloud Datastore'
+    })
+
+    client.put(task)
+    # [END insert]
+
+    return task
+
+
+def update(client):
+    # Create the entity we're going to update.
+    upsert(client)
+
+    # [START update]
+    key = client.key('Task', 'sample_task')
+    task = client.get(key)
+
+    task['done'] = True
+
+    client.put(task)
+    # [END update]
+
+    return task
+
+
+def lookup(client):
+    # Create the entity that we're going to look up.
+    upsert(client)
+
+    # [START lookup]
+    key = client.key('Task', 'sample_task')
+    task = client.get(key)
+    # <Entity
+    #   [{'kind': 'Task', 'name': 'sample_task'}]
+    #   {
+    #       'priority': 4,
+    #       'type': 'Personal',
+    #       'done': False,
+    #       'description': 'Learn Cloud Datastore'
+    # }>
+    # [END lookup]
+
+    return task
+
+
+def delete(client):
+    # Create the entity we're going to delete.
+    upsert(client)
+
+    # [START delete]
+    key = client.key('Task', 'sample_task')
+    client.delete(key)
+    # [END delete]
+
+    return key
+
+
+def batch_upsert(client):
+    # [START batch_upsert]
+    task1 = datastore.Entity(client.key('Task', 1))
+
+    task1.update({
+        'type': 'Personal',
+        'done': False,
+        'priority': 4,
+        'description': 'Learn Cloud Datastore'
+    })
+
+    task2 = datastore.Entity(client.key('Task', 2))
+
+    task2.update({
+        'type': 'Work',
+        'done': False,
+        'priority': 8,
+        'description': 'Integrate Cloud Datastore'
+    })
+
+    client.put_multi([task1, task2])
+    # [END batch_upsert]
+
+    return task1, task2
+
+
+def batch_lookup(client):
+    # Create the entities we will lookup.
+    batch_upsert(client)
+
+    # [START batch_lookup]
+    keys = [
+        client.key('Task', 1),
+        client.key('Task', 2)
+    ]
+
+    tasks = client.get_multi(keys)
+    # [<Entity Task 1>, <Entity Task 2>]
+    # [END batch_lookup]
+
+    return tasks
+
+
+def batch_delete(client):
+    # Create the entities we will delete.
+    batch_upsert(client)
+
+    # [START batch_delete]
+    keys = [
+        client.key('Task', 1),
+        client.key('Task', 2)
+    ]
+
+    client.delete_multi(keys)
+    # [END batch_delete]
+
+    return keys
+
+
+def unindexed_property_query(client):
+    # Create the entity that we're going to query.
+    upsert(client)
+
+    # [START unindexed_property_query]
+    query = client.query(kind='Task')
+    query.add_filter('description', '=', 'Learn Cloud Datastore')
+    tasks = list(query.fetch())
+    # [<Entity Task>, ... ]
+    # [END unindexed_property_query]
+
+    return tasks
+
+
+def basic_query(client):
+    # Create the entity that we're going to query.
+    upsert(client)
+
+    # [START basic_query]
+    query = client.query(kind='Task')
+    query.add_filter('done', '=', False)
+    query.add_filter('priority', '>=', 4)
+    query.order = ['-priority']
+
+    tasks = list(query.fetch())
```

Comment:
basic_query should only define the query, this piece should be in a separate run_query snippet.